### PR TITLE
throw catchable exception instead of exit in JSON response

### PR DESCRIPTION
**Pull request by nexxome**
Hi,
it would be awesome if the JSON response throws an exception on other response status codes instead of exit because then you can catch it and proceed with code execution.

Thanks

### CODE REVIEW

File: src/Response/JSON.php [hunk 1]
```diff
@@ -29,8 +29,11 @@ public function parse($data, array $headers)
                 $headers['Status'] === 422 ||
                 $headers['Status'] === 400
             )) {
-                print_r($headers);
-                exit;
+                throw new Exception([
```
Comment (simonschaufi) [inline src/Response/JSON.php]: don't get it. Who will catch this exception?

**Comment by nexxome** [comment]
The XML response always throws an exception on error, the JSON response doesn't. If I have multiple requests and one will fail, then the application just exits and there is no way to prevent this. If it would throw an exception instead, you can respond to it:

```php
try {
    $response = $timeModel->getByProject($projectId);
}catch (\Exception $e){
    ...
}

**Comment by simonschaufi** [comment]
I know the user can do that but what if he doesn't?

**Comment by nexxome** [comment]
How can anybody rely on a behavior on which you can't respond to?

**Comment by simonschaufi** [comment]
good question. I will then just merge it like that and hope nobody will complain later on :D

**Pull request merged**